saleor/graphql/webhook/subscription_query.py

```diff
@@ -0,0 +1,138 @@
+from enum import Enum, Flag
+from typing import Dict, List, Optional, Union
+
+from django.core.exceptions import ValidationError
+from graphene.utils.str_converters import to_snake_case
+from graphql import get_default_backend, validate
+from graphql.error import GraphQLSyntaxError
+from graphql.language.ast import (
+    Document,
+    Field,
+    FragmentDefinition,
+    FragmentSpread,
+    InlineFragment,
+    OperationDefinition,
+)
+
+
+class IsFragment(Flag):
+    TRUE = True
+    FALSE = False
+
+
+class SubscriptionQueryError(Enum):
+    MISSING_SUBSCRIPTION = "Subscription operation can't be found."
+    MISSING_EVENT_FIELD = "Event field can't be found."
+    MISSING_EVENTS = "Can't find a single event."
+
+
+class SubscriptionQuery:
+    def __init__(self, query: str):
+        self.query: str = query
+        self.is_valid: bool = False
+        self.ast: Document = Document("")
+        self.events: List[str] = []
+        self.errors = self.validate_query()
+        self.error_msg = ";".join([err.message for err in self.errors])
+
+    def validate_query(self):
+        from ..api import schema
+
+        graphql_backend = get_default_backend()
+        try:
+            document = graphql_backend.document_from_string(schema, self.query)
+            self.ast = document.document_ast
+            errors = validate(schema, self.ast)
+        except GraphQLSyntaxError as e:
+            return [e]
+
+        if errors:
+            return errors
+
+        try:
+            self.events = self.get_events_from_subscription()
+        except Exception as err:
+            return [err]
+
+        self.is_valid = True
```

Comment:
This means, that subscription query is valid only if events can be parsed. Any objections?

Reply:
I think it makes sense.